src/python_testing/TC_DISHM_2_1.py

```diff
@@ -0,0 +1,168 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+import chip.clusters as Clusters
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main, type_matches
+from mobly import asserts
+
+# This test requires several additional command line arguments
+# run with
+# --int-arg PIXIT_ENDPOINT:<endpoint> PIXIT_MODEOK:<mode id> PIXIT_MODEFAIL:<mode id>
+
+
+class TC_DISHM_2_1(MatterBaseTest):
+
+    async def read_mod_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.DishwasherMode
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    async def send_change_to_mode_cmd(self, newMode) -> Clusters.Objects.DishwasherMode.Commands.ChangeToModeResponse:
+        ret = await self.send_single_cmd(cmd=Clusters.Objects.DishwasherMode.Commands.ChangeToMode(newMode=newMode), endpoint=self.endpoint)
+        asserts.assert_true(type_matches(ret, Clusters.Objects.DishwasherMode.Commands.ChangeToModeResponse),
+                            "Unexpected return type for ChangeToMode")
+        return ret
+
+    @async_test_body
+    async def test_TC_DISHM_2_1(self):
+
+        asserts.assert_true('PIXIT_ENDPOINT' in self.matter_test_config.global_test_params,
+                            "PIXIT_ENDPOINT must be included on the command line in "
+                            "the --int-arg flag as PIXIT_ENDPOINT:<endpoint>")
+        asserts.assert_true('PIXIT_MODEOK' in self.matter_test_config.global_test_params,
+                            "PIXIT_MODEOK must be included on the command line in "
+                            "the --int-arg flag as PIXIT_MODEOK:<mode id>")
+        asserts.assert_true('PIXIT_MODEFAIL' in self.matter_test_config.global_test_params,
+                            "PIXIT_MODEFAIL must be included on the command line in "
+                            "the --int-arg flag as PIXIT_MODEFAIL:<mode id>")
+        asserts.assert_true('pixit_can_test_mode_failure' in self.matter_test_config.global_test_params,
```

Comment:
So is it `pixit_can_test_mode_failure` or `PIXIT_CAN_TEST_MODE_FAILURE`?

Reply:
Addressed and updated